src/event/metric.rs

```diff
@@ -257,6 +261,21 @@ impl Metric {
     pub fn tag_value(&self, name: &str) -> Option<String> {
         self.tags.as_ref().and_then(|t| t.get(name).cloned())
     }
+
+    /// Sets or updates the string value of a tag
+    pub fn set_tag_value(&mut self, name: String, value: String) {
+        if self.tags.is_none() {
+            let map = BTreeMap::new();
+            self.tags = Some(map);
+        }
+
+        self.tags.as_mut().and_then(|tags| tags.insert(name, value));
```

Comment:
There's a `get_or_insert` on `Option` which should make this a bit easier to write.

Reply:
Doh. Yes, I remember this came up in another review!